server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -220,7 +220,8 @@ protected DirectoryReader open(IndexCommit commit) throws IOException {
         assert Transports.assertNotTransportThread("opening index commit of a read-only engine");
         DirectoryReader directoryReader = DirectoryReader.open(
             commit,
-            org.apache.lucene.util.Version.MIN_SUPPORTED_MAJOR,
+            // Temporary workaround
+            config().getIndexSettings().getIndexMetadata().getCompatibilityVersion().luceneVersion().major,
```

Comment:
Why not use the N-2 version here? Fine as is for now ofc.

Reply:
Sure, I changed that